web/packages/teleterm/src/services/tshd/createClient.ts

```diff
@@ -17,19 +17,25 @@
  */
 
 import { GrpcTransport } from '@protobuf-ts/grpc-transport';
-import { TerminalServiceClient } from 'gen-proto-ts/teleport/lib/teleterm/v1/service_pb.client';
+import {
+  ITerminalServiceClient,
+  TerminalServiceClient,
+} from 'gen-proto-ts/teleport/lib/teleterm/v1/service_pb.client';
 import * as vnetServiceProtobuf from 'gen-proto-ts/teleport/lib/teleterm/vnet/v1/vnet_service_pb.client';
 
 import { CloneableClient, cloneClient } from './cloneableClient';
-import * as types from './types';
 
-export function createTshdClient(transport: GrpcTransport): types.TshdClient {
+// Creating the client type based on the interface (ITerminalServiceClient) and not the class
+// (TerminalServiceClient) lets us omit a bunch of properties when mocking a client.
+export type TshdClient = CloneableClient<ITerminalServiceClient>;
+
+export type VnetClient =
+  CloneableClient<vnetServiceProtobuf.IVnetServiceClient>;
+
+export function createTshdClient(transport: GrpcTransport): TshdClient {
   return cloneClient(new TerminalServiceClient(transport));
 }
 
-export function createVnetClient(transport: GrpcTransport): VnetServiceClient {
+export function createVnetClient(transport: GrpcTransport): VnetClient {
   return cloneClient(new vnetServiceProtobuf.VnetServiceClient(transport));
 }
```

Comment:
Nit: Just for consistency with `TerminalServiceClient`, I'd import the service client as 
```ts
import {
  IVnetServiceClient,
  VnetServiceClient,
} from 'gen-proto-ts/teleport/lib/teleterm/vnet/v1/vnet_service_pb.client';
```
instead of `vnetServiceProtobuf`.